include/EASTL/tuple.h

```diff
@@ -145,6 +145,44 @@ class tuple_element<I, const volatile Internal::TupleImpl<Indices, Ts...>> : pub
 {
 };
 
+// attempt to isolate index given a type
+template <typename T, typename Tuple>
+struct tuple_index
+{
+};
+
+template <typename T>
+struct tuple_index<T, TupleTypes<>>
+{
+	typedef void DuplicateTypeCheck;
+	tuple_index() = delete; // tuple_index should only be used for compile-time assistance, and never be instantiated
+	static const size_t index = 0;
+};
+
+template <typename T, typename... TsRest>
+struct tuple_index<T, TupleTypes<T, TsRest...>>
+{
+	typedef int DuplicateTypeCheck;
+	// after finding type T in the list of types, try to find type T in TsRest.
+	// If we stumble back into this version of tuple_index, i.e. type T appears twice in the list of types, then DuplicateTypeCheck will be of type int, and the statiC_assert will fail.
```

Comment:
No wonder I didn't eyeball that; the difference between capital and lowercase in my ide's font in this case is really hard to see.
![image](https://cloud.githubusercontent.com/assets/19677493/19875446/26bb2032-9fa3-11e6-9150-5658fc449842.png)